cfpb/sbl-frontend
# Update financial institution profile - Phase1 - Gather inputs

## Changes
- Pre-populates fields with existing Institution data (Phase 3)
- On submit, gathers user input for editable fields and currently just prints to console (Phase 1)(thanks Sigmund!)
- <img width="443" alt="Screenshot 2024-02-12 at 4 26 33 PM" src="https://github.com/cfpb/sbl-frontend/assets/2592907/0d54ba25-3cbb-4eeb-b637-43b0cf2f84b3">
## How to test
- Visit the [Update your financial institution profile](http://localhost:8899/institution/TESTBANK123/update) page
- Change some data
- Open the Inspector panel
- Hit the `Submit` button
- Observe the form data being printed out to the console (similar to the screenshot above)


## Questions
- [x] Think I recall devs saying we wanted to send *all* institution data, but that seems wasteful.  Seems more sensible to only send the fields that the user has changed.  That way SBL Help does not need to parse out the difference, they'll be able to focus on the handful of fields with actual changes. 

  To be revisted: https://github.com/cfpb/sbl-frontend/issues/253
- [ ] Do we need to establish the pattern for how we name fields right now? Or can that wait until we have the API endpoint and are actually sending data somewhere? i.e. The `additional_details` field in the screenshot above is associated with `institutionType.other` checkbox - but that is not currently clear. 
- [ ] What do folks think about the approach of re-using some of the View components as Update components? ex. Since on the update page there is only 1 editable field, I've modified the [View page's AffiliateInformation](https://github.com/cfpb/sbl-frontend/blob/4ac104fe5c57cf39895929c1511fb2dc854f1be4/src/pages/Filing/ViewInstitutionProfile/AffiliateInformation.tsx) component to accept an `isUpdate` flag so that I can reuse the all of the display aspects of the component.  This is probably a major violation of "single responsibility" but saved me from reimplementing some stuff.  


## Screenshots
- ![screencapture-localhost-8899-institution-TESTBANK123-update-2024-02-12-16_53_54](https://github.com/cfpb/sbl-frontend/assets/2592907/4bed44c0-a4f8-4261-9332-baf4ff032747)


## TODO (maybe)
- Lots of TODOs in these files related to ESLint (maybe these are things to be addressed in Phase 4, where we implement validation?)
